### PR TITLE
Prow monitoring cleanup, codegen, deps, and deployment

### PR DESCRIPTION
**Pull request by clarketm**
follow up items for: https://github.com/istio/test-infra/issues/1774

blocked by:
- [x] Fix slack webhook api url.
- [x] Add dependencies to `build-tools` image (e.g. `jsonnet`, `jb`). (https://github.com/istio/tools/pull/869)
- [x] Setup DNS for `monitoring.prow.istio.io`
- [x] Create ProwJob to autodeploy.
- [x] Manually create/deploy `alertmanager-prow_secret.yaml` and `grafana_secret.yaml` with secret values.

### CODE REVIEW

**Review by scottilee — APPROVED** [review]
LGTM!

File: prow/config/jobs/test-infra.yaml [hunk 1]
```diff
@@ -33,22 +33,62 @@ jobs:
...
     - deploy
-    - deploy-gcsweb
-    - deploy-velodrome
     - deploy-build
     - deploy-private
     requirements: [deploy]
     node_selector:
       prod: prow
 
+  - name: deploy-gcsweb
```
Comment (fejta) [inline prow/config/jobs/test-infra.yaml]: Can we just put all of this in the deploy prow job please? AKA just as we do not need a separate mechanism to deploy crier outside of deploy prow, we do not need a separate mechanism to deploy prow's monitoring.

**Review by fejta — APPROVED** [review]
/hold
merge at will

**Review by chases2 — APPROVED** [review]

**Pull request merged**